src/tracing/registry.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <lib/support/IntrusiveList.h>
+
+#include <platform/LockTracker.h>
+#include <tracing/registry.h>
+
+namespace chip {
+namespace Tracing {
+namespace {
+
+IntrusiveList<Backend> gTracingBackends;
+
+} // namespace
+
+void Register(Backend & backend)
+{
+    assertChipStackLockedByCurrentThread();
+    gTracingBackends.PushBack(&backend);
+}
+
+void Unregister(Backend & backend)
+{
+    assertChipStackLockedByCurrentThread();
+    gTracingBackends.Remove(&backend);
+}
+
+#ifdef MATTER_TRACING_ENABLED
+
+namespace Internal {
+
+void Begin(::chip::Tracing::Scope scope)
+{
+    for (auto backend = gTracingBackends.begin(); backend != gTracingBackends.end(); backend++)
```

Comment:
Any reason not to `for (auto & backend : gTracingBackends)` ?

Reply:
+1

Reply:
This is what I wrote at first and I got compiler errors I belive .... will try it again to check...

Reply:
This is what I wrote at first and I got compiler errors I belive .... will try it again to check...

Reply:
I had used the begin/end and that fails:

```
2023-06-01 13:19:36 INFO    ../../src/tracing/registry.cpp:48:49: error: cannot bind non-const lvalue reference of type ‘chip::IntrusiveList<chip::Tracing::Backend>::Iterator&’ to an rvalue of type ‘chip::IntrusiveList<chip::Tracing::Backend>::Iterator’
2023-06-01 13:19:36 INFO       48 |     for (auto & backend = gTracingBackends.begin(); backend != gTracingBackends.end(); backend++)
2023-06-01 13:19:36 INFO          |  
```

however your form works. Updating.